File: php/create/Factory.php
<?php

class ACar
{
    public function run()
    {
        echo 'ACar run',"\n";
    }
}

class BCar
{
    public function run()
    {
        echo 'BCar run',"\n";
    }
}

class CarFactory
{
    public function getCar($name)
    {
        return new $name();
    }
}

class Client
{
    public function do()
    {
        $f = new CarFactory();
        $acar = $f->getCar('ACar');
        $acar->run();
    }
}

$c = new Client();
$c->do();

File: ruby/create/Factory.rb
module Factory1
    class Car
        attr_accessor :name
        def initialize
        end
        def run
            puts "#{@name} is runing"
        end
    end
    
    class Acar < Car
        def initialize
            @name = 'Acar'
        end
    end
    class Bcar < Car
        def initialize
            @name = 'Bcar'
        end
    end

    class Factory
        def initialize
            
        end
        def getCar(carType)
            Factory1.const_get(carType).new
        end
    end

    module Test
        f = Factory.new
        c1 = f.getCar("Acar")
        c2 = f.getCar("Bcar")

        c1.run
        c2.run

    end
end